### PR TITLE
feat(what-input): Limit keyboard detection in inputs

### PR DESCRIPTION
**Pull request by jurokapsiar**
## Current Behavior
what-input is listening to keyboard, mouse and touch events and switches input mode. This change is stored to `body` element using `data-what-input` attribute. Fluent generates classes for :focus-visible pseudo selector that get activated when keyboard mode is used. This however causes a large style recalc.

## New Behavior

Keyboard detection is ignored if user is in an input field and the key is not tab or F6.
Input fields are by design designed in a way where they do not need specific keyboard focused styles. We only need to check if the mode changed as user navigates outside of the input.

## Related Issue(s)

Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/443c503b94e602e55857e4e311413c6257d60766'>443c503b94e602e55857e4e311413c6257d60766</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## Perf Analysis (`@fluentui/react-northstar`)
<details><summary>Perf tests with no regressions</summary>

Scenario | Current PR Ticks | Baseline Ticks | Ratio
:--- | ---:| ---:| ---:
ButtonMinimalPerf.default | [143](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ButtonMinimalPerf.default.html) | [132](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ButtonMinimalPerf.default_base.html) | 1.08:1
RefMinimalPerf.default | [190](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/RefMinimalPerf.default.html) | [177](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/RefMinimalPerf.default_base.html) | 1.07:1
AttachmentMinimalPerf.default | [134](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AttachmentMinimalPerf.default.html) | [128](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AttachmentMinimalPerf.default_base.html) | 1.05:1
SegmentMinimalPerf.default | [306](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/SegmentMinimalPerf.default.html) | [291](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/SegmentMinimalPerf.default_base.html) | 1.05:1
TreeWith60ListItems.default | [140](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TreeWith60ListItems.default.html) | [133](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TreeWith60ListItems.default_base.html) | 1.05:1
LayoutMinimalPerf.default | [323](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/LayoutMinimalPerf.default.html) | [312](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/LayoutMinimalPerf.default_base.html) | 1.04:1
PortalMinimalPerf.default | [146](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/PortalMinimalPerf.default.html) | [140](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/PortalMinimalPerf.default_base.html) | 1.04:1
TableMinimalPerf.default | [366](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TableMinimalPerf.default.html) | [352](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TableMinimalPerf.default_base.html) | 1.04:1
BoxMinimalPerf.default | [310](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/BoxMinimalPerf.default.html) | [301](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/BoxMinimalPerf.default_base.html) | 1.03:1
FormMinimalPerf.default | [335](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/FormMinimalPerf.default.html) | [324](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/FormMinimalPerf.default_base.html) | 1.03:1
ListCommonPerf.default | [511](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ListCommonPerf.default.html) | [496](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ListCommonPerf.default_base.html) | 1.03:1
MenuButtonMinimalPerf.default | [1370](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/MenuButtonMinimalPerf.default.html) | [1326](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/MenuButtonMinimalPerf.default_base.html) | 1.03:1
IconMinimalPerf.default | [573](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/IconMinimalPerf.default.html) | [559](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/IconMinimalPerf.default_base.html) | 1.03:1
AvatarMinimalPerf.default | [163](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AvatarMinimalPerf.default.html) | [160](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AvatarMinimalPerf.default_base.html) | 1.02:1
FlexMinimalPerf.default | [254](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/FlexMinimalPerf.default.html) | [250](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/FlexMinimalPerf.default_base.html) | 1.02:1
ImageMinimalPerf.default | [346](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ImageMinimalPerf.default.html) | [339](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ImageMinimalPerf.default_base.html) | 1.02:1
ListMinimalPerf.default | [461](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ListMinimalPerf.default.html) | [452](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ListMinimalPerf.default_base.html) | 1.02:1
PopupMinimalPerf.default | [564](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/PopupMinimalPerf.default.html) | [554](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/PopupMinimalPerf.default_base.html) | 1.02:1
SliderMinimalPerf.default | [1255](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/SliderMinimalPerf.default.html) | [1235](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/SliderMinimalPerf.default_base.html) | 1.02:1
ToolbarMinimalPerf.default | [793](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ToolbarMinimalPerf.default.html) | [781](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ToolbarMinimalPerf.default_base.html) | 1.02:1
AlertMinimalPerf.default | [223](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AlertMinimalPerf.default.html) | [221](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AlertMinimalPerf.default_base.html) | 1.01:1
CardMinimalPerf.default | [467](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/CardMinimalPerf.default.html) | [462](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/CardMinimalPerf.default_base.html) | 1.01:1
CheckboxMinimalPerf.default | [1538](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/CheckboxMinimalPerf.default.html) | [1520](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/CheckboxMinimalPerf.default_base.html) | 1.01:1
DialogMinimalPerf.default | [690](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DialogMinimalPerf.default.html) | [684](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DialogMinimalPerf.default_base.html) | 1.01:1
DividerMinimalPerf.default | [314](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DividerMinimalPerf.default.html) | [311](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DividerMinimalPerf.default_base.html) | 1.01:1
DropdownManyItemsPerf.default | [532](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DropdownManyItemsPerf.default.html) | [529](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DropdownManyItemsPerf.default_base.html) | 1.01:1
GridMinimalPerf.default | [293](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/GridMinimalPerf.default.html) | [290](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/GridMinimalPerf.default_base.html) | 1.01:1
HeaderSlotsPerf.default | [686](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/HeaderSlotsPerf.default.html) | [681](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/HeaderSlotsPerf.default_base.html) | 1.01:1
ItemLayoutMinimalPerf.default | [979](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ItemLayoutMinimalPerf.default.html) | [968](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ItemLayoutMinimalPerf.default_base.html) | 1.01:1
LoaderMinimalPerf.default | [521](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/LoaderMinimalPerf.default.html) | [517](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/LoaderMinimalPerf.default_base.html) | 1.01:1
ReactionMinimalPerf.default | [334](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ReactionMinimalPerf.default.html) | [332](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ReactionMinimalPerf.default_base.html) | 1.01:1
ButtonOverridesMissPerf.default | [1024](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ButtonOverridesMissPerf.default.html) | [1020](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ButtonOverridesMissPerf.default_base.html) | 1:1
ButtonSlotsPerf.default | [431](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ButtonSlotsPerf.default.html) | [431](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ButtonSlotsPerf.default_base.html) | 1:1
ChatDuplicateMessagesPerf.default | [214](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ChatDuplicateMessagesPerf.default.html) | [215](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ChatDuplicateMessagesPerf.default_base.html) | 1:1
DropdownMinimalPerf.default | [2166](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DropdownMinimalPerf.default.html) | [2167](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DropdownMinimalPerf.default_base.html) | 1:1
EmbedMinimalPerf.default | [2645](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/EmbedMinimalPerf.default.html) | [2651](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/EmbedMinimalPerf.default_base.html) | 1:1
InputMinimalPerf.default | [844](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/InputMinimalPerf.default.html) | [841](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/InputMinimalPerf.default_base.html) | 1:1
RadioGroupMinimalPerf.default | [391](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/RadioGroupMinimalPerf.default.html) | [391](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/RadioGroupMinimalPerf.default_base.html) | 1:1
SkeletonMinimalPerf.default | [303](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/SkeletonMinimalPerf.default.html) | [304](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/SkeletonMinimalPerf.default_base.html) | 1:1
SplitButtonMinimalPerf.default | [3266](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/SplitButtonMinimalPerf.default.html) | [3267](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/SplitButtonMinimalPerf.default_base.html) | 1:1
StatusMinimalPerf.default | [601](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/StatusMinimalPerf.default.html) | [600](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/StatusMinimalPerf.default_base.html) | 1:1
CustomToolbarPrototype.default | [2176](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/CustomToolbarPrototype.default.html) | [2166](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/CustomToolbarPrototype.default_base.html) | 1:1
VideoMinimalPerf.default | [622](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/VideoMinimalPerf.default.html) | [620](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/VideoMinimalPerf.default_base.html) | 1:1
AnimationMinimalPerf.default | [466](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AnimationMinimalPerf.default.html) | [471](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AnimationMinimalPerf.default_base.html) | 0.99:1
CarouselMinimalPerf.default | [348](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/CarouselMinimalPerf.default.html) | [351](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/CarouselMinimalPerf.default_base.html) | 0.99:1
ChatMinimalPerf.default | [626](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ChatMinimalPerf.default.html) | [632](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ChatMinimalPerf.default_base.html) | 0.99:1
ChatWithPopoverPerf.default | [289](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ChatWithPopoverPerf.default.html) | [292](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ChatWithPopoverPerf.default_base.html) | 0.99:1
LabelMinimalPerf.default | [332](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/LabelMinimalPerf.default.html) | [337](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/LabelMinimalPerf.default_base.html) | 0.99:1
ListNestedPerf.default | [460](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ListNestedPerf.default.html) | [464](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ListNestedPerf.default_base.html) | 0.99:1
MenuMinimalPerf.default | [728](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/MenuMinimalPerf.default.html) | [737](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/MenuMinimalPerf.default_base.html) | 0.99:1
RosterPerf.default | [1687](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/RosterPerf.default.html) | [1709](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/RosterPerf.default_base.html) | 0.99:1
ProviderMergeThemesPerf.default | [985](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ProviderMergeThemesPerf.default.html) | [996](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ProviderMergeThemesPerf.default_base.html) | 0.99:1
TableManyItemsPerf.default | [1560](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TableManyItemsPerf.default.html) | [1576](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TableManyItemsPerf.default_base.html) | 0.99:1
TextMinimalPerf.default | [294](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TextMinimalPerf.default.html) | [296](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TextMinimalPerf.default_base.html) | 0.99:1
TextAreaMinimalPerf.default | [395](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TextAreaMinimalPerf.default.html) | [398](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TextAreaMinimalPerf.default_base.html) | 0.99:1
TreeMinimalPerf.default | [685](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TreeMinimalPerf.default.html) | [689](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TreeMinimalPerf.default_base.html) | 0.99:1
AttachmentSlotsPerf.default | [866](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AttachmentSlotsPerf.default.html) | [883](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AttachmentSlotsPerf.default_base.html) | 0.98:1
HeaderMinimalPerf.default | [308](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/HeaderMinimalPerf.default.html) | [313](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/HeaderMinimalPerf.default_base.html) | 0.98:1
ProviderMinimalPerf.default | [313](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ProviderMinimalPerf.default.html) | [320](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ProviderMinimalPerf.default_base.html) | 0.98:1
DatepickerMinimalPerf.default | [4508](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DatepickerMinimalPerf.default.html) | [4648](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/DatepickerMinimalPerf.default_base.html) | 0.97:1
TooltipMinimalPerf.default | [1825](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TooltipMinimalPerf.default.html) | [1883](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/TooltipMinimalPerf.default_base.html) | 0.97:1
ListWith60ListItems.default | [485](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ListWith60ListItems.default.html) | [510](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/ListWith60ListItems.default_base.html) | 0.95:1
AccordionMinimalPerf.default | [120](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AccordionMinimalPerf.default.html) | [130](https://fluentuipr.z22.web.core.windows.net/pull/25087/perf-test-northstar/AccordionMinimalPerf.default_base.html) | 0.92:1

</details><div id="perfComment9424"/>

File: packages/fluentui/react-northstar/src/utils/whatInput.ts [hunk 1]
```diff
@@ -108,6 +108,20 @@ const addListeners = (eventTarget: Window) => {
   eventTarget.addEventListener('keyup', eventBuffer, true);
 };
 
+const ignoreKeyboardFocused = (document: Document, eventKey: number) => {
```
Comment (ling1726) [inline packages/fluentui/react-northstar/src/utils/whatInput.ts:111]: I would pick a clearer name for this: `keyboardInputFocused` ?

Also the function definitely needs some docstring to explain what is going on and why this is the case

**Review by ling1726 — APPROVED** [review]
Please add a PR desecription, about the motivation of this PR

**Pull request merged**